com.unity.render-pipelines.universal/Editor/Converter/ReadonlyMaterialConverter.cs

```diff
@@ -0,0 +1,275 @@
+                })
+                    .ToList();
+
+                foreach (var r in orderedRequest)
+                {
+                    if (r == null || !GlobalObjectId.TryParse(r.id, out var gid))
+                    {
+                        continue;
+                    }
+
+                    var label = r.provider.fetchLabel(r, r.context);
+                    var description = r.provider.fetchDescription(r, r.context);
+
+                    var item = new ConverterItemDescriptor()
+                    {
+                        name = $"{label} : {description}",
+                        info = gid.ToString(),
+                    };
+
+                    ctx.AddAssetToConvert(item);
+                }
+
+                callback.Invoke();
+            });
+        }
+
+        public override void OnRun(ref RunItemContext ctx)
+        {
+            // order by the assetGuid so that they run in order
+            //var items = ctx.items.ToList();
+            //var items = ctx.item.ToList();
+            List<ConverterItemInfo> items = new List<ConverterItemInfo>();
+            items.Add(ctx.item);
+            foreach (var(index, obj) in EnumerateObjects(items, ctx).Where(item => item != null))
+            {
+                var materials = MaterialReferenceBuilder.GetMaterialsFromObject(obj);
+
+                var result = true;
+                var errorString = new StringBuilder();
+                foreach (var material in materials)
+                {
+                    // there might be multiple materials on this object, we only care about the ones we explicitly try to remap that fail
+                    if (!MaterialReferenceBuilder.GetIsReadonlyMaterial(material)) continue;
+                    if (!ReadonlyMaterialMap.Map.ContainsKey(material.name)) continue;
+                    if (!ReAssignMaterial(obj, material.name, ReadonlyMaterialMap.Map[material.name]))
+                    {
+                        result = false;
+                        errorString.AppendLine($"Material {material.name} failed to be reassigned");
+                    }
+                }
+
+                if (!result)
+                {
+                    ctx.didFail = true;
+                    ctx.info = errorString.ToString();
+                }
+            }
+        }
+
+        private static bool ReAssignMaterial(Object obj, string oldMaterialName, string newMaterialPath)
+        {
+            var result = false;
+
+            // do the reflection to make sure we get the right material reference
+            if (obj is GameObject go)
+            {
+                foreach (var key in MaterialReferenceBuilder.GetComponentTypes())
+                {
+                    var components = go.GetComponentsInChildren(key);
+                    foreach (var component in components)
+                    {
+                        result = ReassignMaterialOnComponentOrObject(component,
```

Comment:
I have updated this logic based on the suggestions. Now it will fail if the expected new material is null or if the setmethod is null, as both cases are expected to succeed. This also allowed me to remove the result from the second method.